CMakeLists.txt

```diff
@@ -38,7 +38,8 @@ rapids_cmake_build_type(Release)
 # ##################################################################################################
 # * build options ----------------------------------------------------------------------------------
 
-option(USE_NVTX "Build with NVTX support" ON)
+option(USE_NVTX "Legacy NVTX control (used if RMM_USE_NVTX is not set)" ON)
+option(RMM_USE_NVTX "Build RMM with NVTX support" ${USE_NVTX})
```

Comment:
We should just remove `USE_NVTX` instead of marking it as legacy.

Consider the following:

```
cmake -S <rmm> -B <build>
cmake -S <rmm> -B <build> -DRMM_USE_NVTX=OFF
```

Since the user only changed `RMM_USE_NVTX`, the `USE_NVTX` will still be enabled and therefore they get nvtx. Likewise if the user just disabled `USE_NVTX` on the second pass it would have zero impact since `RMM_USE_NVTX` was already created on the first pass and therefore is still enabled.

Since you need to mark both `USE_NVTX` and `RMM_USE_NVTX` as `OFF` to actually disable NVTX support I see no need for both flags as it is just a foot gun.

Reply:
We are going to have to break backwards compat at some point. Lets do it now and not introduce a source of bugs between 24.08 and the backwards compat break.

Reply:
I removed the use of USE_NVTX.  Note that this means projects that expected NVTX ranges to be disabled for everything via -DUSE_NVTX=OFF as before will still leave RMM NVTX ranges enabled.